core/src/main/java/tech/ydb/core/impl/auth/GrpcAuthRpc.java

```diff
@@ -42,7 +43,7 @@ public String getDatabase() {
     }
 
     public void changeEndpoint() {
-        endpointIdx = (endpointIdx + 1) % endpoints.size();
+        endpointIdx.getAndIncrement();
```

Comment:
What is happens when we call changeEndpoint() Integer.MAX_VALUE times?

Reply:
Are you sure that we will achieve this?